src/btchip_apdu_sign_message.c

```diff
@@ -262,7 +262,7 @@ unsigned short btchip_compute_hash() {
         goto discard;
     }
             
-    if (cx_hash_sha256(hash, sizeof(hash), hash, 32)) {
+    if (!cx_hash_sha256(hash, sizeof(hash), hash, 32)) {
```

Comment:
```suggestion
    if (cx_hash_sha256(hash, sizeof(hash), hash, 32) == 0) {
```
Avoiding implicit int / bool cast is a good way to prevent this type of issue